src/configurationProvider.ts

```diff
@@ -86,44 +88,56 @@ export class JavaDebugConfigurationProvider implements vscode.DebugConfiguration
+                type: "java",
+                name: "Debug (Launch) - Current File",
+                request: "launch",
+                // tslint:disable-next-line
+                mainClass: "${file}",
+            };
+            try {
+                const isOnStandardMode = await utility.waitForStandardMode(progressReporter);
+                if (!isOnStandardMode) {
+                    resolve([defaultLaunchConfig]);
+                    return ;
+                }
 
-                    const mainClasses = await lsPlugin.resolveMainClass(folder ? folder.uri : undefined);
-                    let cache: {[key: string]: any};
-                    cache = {};
-                    const launchConfigs = mainClasses.map((item) => {
-                        return {
-                            ...defaultLaunchConfig,
-                            name: this.constructLaunchConfigName(item.mainClass, cache, item.projectName),
-                            mainClass: item.mainClass,
-                            projectName: item.projectName,
-                        };
-                    });
-                    resolve([defaultLaunchConfig, ...launchConfigs]);
-                } catch (ex) {
-                    if (ex instanceof utility.JavaExtensionNotEnabledError) {
-                        utility.guideToInstallJavaExtension();
-                    }
-                    p.report({ message: `failed to generate configuration. ${ex}` });
-                    resolve(defaultLaunchConfig);
+                if (progressReporter.isCancelled()) {
+                    resolve([defaultLaunchConfig]);
+                    return;
                 }
-            });
+                progressReporter.report("Resolve Java Configs", "Auto generating Java configuration...");
+                const mainClasses = await lsPlugin.resolveMainClass(folder ? folder.uri : undefined);
+                const cache = {};
+                const launchConfigs = mainClasses.map((item) => {
+                    return {
+                        ...defaultLaunchConfig,
+                        name: this.constructLaunchConfigName(item.mainClass, cache, item.projectName),
+                        mainClass: item.mainClass,
+                        projectName: item.projectName,
+                    };
+                });
+                if (progressReporter.isCancelled()) {
+                    resolve([defaultLaunchConfig]);
+                    return;
+                }
+                resolve([defaultLaunchConfig, ...launchConfigs]);
+            } catch (ex) {
+                if (ex instanceof utility.JavaExtensionNotEnabledError) {
+                    utility.guideToInstallJavaExtension();
+                } else {
+                    // tslint:disable-next-line
+                    console.error(ex);
+                }
+
+                resolve([defaultLaunchConfig]);
+            } finally {
+                progressReporter.cancel();
```

Comment:
Does that mean that the invocation of `progressReporter.cancel()` is required as its lifecycle method? If that is the case, maybe calling it as `.done()` is better.

Reply:
`.done()` makes sense.